src/Parser/Hunter/Shared/Modules/Features/FocusUsage.js

```diff
@@ -0,0 +1,323 @@
+                display: 'inline-block',
+                background: color,
+                borderRadius: '50%',
+                width: 16,
+                height: 16,
+                marginBottom: -3,
+              }}
+            />
+          </div>
+          <div className="flex-main" style={{ paddingLeft: 5 }}>
+            {label}
+          </div>
+          <div className="flex-sub">
+            <dfn data-tip={value}>
+              {formatPercentage(value / total, 1)}%
+            </dfn>
+          </div>
+        </div>
+      );
+    });
+  }
+  chart(items) {
+    return (
+      <DoughnutChart
+        data={{
+          datasets: [{
+            data: items.map(item => item.value),
+            backgroundColor: items.map(item => item.color),
+            borderColor: '#666',
+            borderWidth: 1.5,
+          }],
+          labels: items.map(item => item.label),
+        }}
+        options={{
+          legend: {
+            display: false,
+          },
+          tooltips: {
+            bodyFontSize: 8,
+          },
+          cutoutPercentage: 45,
+          animation: false,
+          responsive: false,
+        }}
+        width={CHART_SIZE}
+        height={CHART_SIZE}
+      />
+    );
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (this.combatants.selected.spec === SPECS.MARKSMANSHIP_HUNTER) {
+      if (LIST_OF_MM_FOCUS_SPENDERS.every(id => spellId !== id)) {
+        return;
+      }
+      this.focusSpenderCastsMM[spellId].casts += 1;
+      this.focusSpenderCastsMM[spellId].focusUsed += event.classResources[0]['cost'] || 0;
+    } else if (this.combatants.selected.spec === SPECS.BEAST_MASTERY_HUNTER) {
+      if (LIST_OF_BM_FOCUS_SPENDERS.every(id => spellId !== id)) {
+        return;
+      }
+      this.focusSpenderCastsBM[spellId].casts += 1;
+      this.focusSpenderCastsBM[spellId].focusUsed += event.classResources[0]['cost'] || 0;
+    }
+  }
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.VOLLEY_ACTIVATED.id) {
+      return;
+    }
+    if (event.timestamp > (this.lastVolleyHit + BUFFER_MS)) {
+      if (this.combatants.selected.spec === SPECS.MARKSMANSHIP_HUNTER) {
+        this.focusSpenderCastsMM[spellId].casts += 1;
+        this.focusSpenderCastsMM[spellId].focusUsed += VOLLEY_FOCUS_COST;
+
+      }
+      if (this.combatants.selected.spec === SPECS.BEAST_MASTERY_HUNTER) {
+        this.focusSpenderCastsBM[spellId].casts += 1;
+        this.focusSpenderCastsBM[spellId].focusUsed += VOLLEY_FOCUS_COST;
+      }
+      this.lastVolleyHit = event.timestamp;
+    }
+  }
+
+  focusUsageChart() {
+    let totalFocusUsed = 0;
+    const items = [];
+    const bmFocusSpenders = [[SPELLS.COBRA_SHOT.id], [SPELLS.MULTISHOT.id], [SPELLS.KILL_COMMAND.id], [SPELLS.REVIVE_PET_AND_MEND_PET.id], [SPELLS.A_MURDER_OF_CROWS_TALENT_SHARED.id], [SPELLS.BARRAGE_TALENT.id], [SPELLS.VOLLEY_ACTIVATED.id]];
```

Comment:
Does every entry need to be an array? I'm having a hard time figuring out why you need this, and I'm thinking your code below might not be working as you're expecting it to.